### DISCUSSION TITLE
error with smart chat

### DISCUSSION THREAD

**Discussion by momofox123** [brianpetro/obsidian-smart-connections#813] [General]
hi i'm on windows, and the latest version of smartconnections and obsidian. 

and I am getting this error when i use the smart chat. may I ask how can I fix it?

TypeError: Cannot read properties of undefined (reading 'embed_batch')
    at SmartEmbedModel.embed_batch (plugin:smart-connections:8916:31)
    at SmartBlocks.lookup (plugin:smart-connections:7204:45)
    at lookup (plugin:smart-connections:12939:27)
    at ScChatModel.handle_tool_call (plugin:smart-connections:3990:35)
    at async ScChatModel.complete (plugin:smart-connections:3905:18)
    at async ScChat.new_user_message (plugin:smart-connections:5204:9)
    at async ScChat.new_user_message (plugin:smart-connections:12701:7)
    
   
![image](https://github.com/user-attachments/assets/dc8502d8-ef61-4481-8a8c-d27b5fee2f77)

**Comment by nicholaswgd** [comment]
> hi i'm on windows, and the latest version of smartconnections and obsidian.
> 
> and I am getting this error when i use the smart chat. may I ask how can I fix it?
> 
> TypeError: Cannot read properties of undefined (reading 'embed_batch') at SmartEmbedModel.embed_batch (plugin:smart-connections:8916:31) at SmartBlocks.lookup (plugin:smart-connections:7204:45) at lookup (plugin:smart-connections:12939:27) at ScChatModel.handle_tool_call (plugin:smart-connections:3990:35) at async ScChatModel.complete (plugin:smart-connections:3905:18) at async ScChat.new_user_message (plugin:smart-connections:5204:9) at async ScChat.new_user_message (plugin:smart-connections:12701:7)
> 
> ![image](https://private-user-images.githubusercontent.com/109475743/372086317-dc8502d8-ef61-4481-8a8c-d27b5fee2f77.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg4NTk3MzcsIm5iZiI6MTcyODg1OTQzNywicGF0aCI6Ii8xMDk0NzU3NDMvMzcyMDg2MzE3LWRjODUwMmQ4LWVmNjEtNDQ4MS04YThjLWQyN2I1ZmVlMmY3Ny5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAxM1QyMjQzNTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jM2JjYjM5ZTlhYjcxNjg4YzBkNTUzZTYyYTcxMWVlNTQyZGExNTYxNzAzNDAzNWQ5ZDg0MzBlYmVlY2JlOWMzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.aDfn-FGbHCpq7thBQXeDKxIpOV_WWXyHX0SLKKp4GlI)

Hi, did you manage to solve it?
